apps/api/src/user/routes/user-api-keys/user-api-keys.router.ts

```diff
@@ -0,0 +1,208 @@
+import { createRoute } from "@hono/zod-openapi";
+import { container } from "tsyringe";
+import { z } from "zod";
+
+import { OpenApiHonoHandler } from "@src/core/services/open-api-hono-handler/open-api-hono-handler";
+import { UserApiKeyController } from "@src/user/controllers/user-api-keys/user-api-keys.controller";
+import {
+  CreateUserApiKeyRequestSchema,
+  FindUserApiKeyParamsSchema,
+  UpdateUserApiKeyRequestSchema,
+  UserApiKeyResponseSchema
+} from "@src/user/http-schemas/user-api-key.schema";
+
+const listRoute = createRoute({
+  method: "get",
+  path: "/v1/users/{userId}/api-keys",
+  summary: "List all API keys",
+  tags: ["User API Keys"],
+  request: {
+    params: z.object({
+      userId: z.string().uuid()
+    })
+  },
+  responses: {
+    200: {
+      description: "Returns list of API keys",
+      content: {
+        "application/json": {
+          schema: z.object({
+            data: z.array(UserApiKeyResponseSchema)
+          })
+        }
+      }
+    }
+  }
+});
+
+const getRoute = createRoute({
+  method: "get",
+  path: "/v1/users/{userId}/api-keys/{id}",
```

Comment:
this kind of path has certain drawbacks:
- harder to manage relationships between resources
- harder to enhance filtering
- harder to read
- harder to unify interfaces across the api

A better approach it to always try to treat resources as top level resources. In this case it should be
```suggestion
  path: "/v1/api-keys/{id}",
```

`userId` parameter is not needed in this case, permissions will handle the use case

Reply:
I would like to ask few question:
1. Should there be an option to display API keys for all users?
2. Should there even be a possibility for one user to fetch another user's keys?

In both cases, neither `/api_keys` nor `/users/{user_id}/api_keys` prevents a potential mistake in the code (missed userId filter). Though option with `/users/{user_id}/api_keys` opens a possibility to guess `userId` and try to fetch api_keys of another user

Reply:
My reasoning for using `/v1/users` prefix to paths was to separate routes at the root to under the "users" sub-routes for the module, or `/v1/auth` following your comment. I don't mind putting it as `/v1/api-keys/...`

> Should there be an option to display API keys for all users?
For an admin panel? I don't think we need that right and we can add it later when needed.

> Should there even be a possibility for one user to fetch another user's keys?
Obviously no, we do validate with the user token and I added tests for this, but it makes sense to remove it from the path